tsl/src/continuous_aggs/options.c

```diff
@@ -62,12 +66,163 @@ update_materialized_only(ContinuousAgg *agg, bool materialized_only)
 	ts_scan_iterator_close(&iterator);
 }
 
+/*
+ * Retrieve the cagg view query and find the groupby clause and
+ * time_bucket clause. Map them to the column names(of mat.hypertable)
+ * Returns: list of column names used in group by clause of the cagg query.
+ */
+static List *
+cagg_find_groupingcols(ContinuousAgg *agg, Hypertable *mat_ht)
+{
+	List *retlist = NIL;
+	ListCell *lc;
+	Oid cagg_view_oid =
+		get_relname_relid(NameStr(agg->data.user_view_name),
+						  get_namespace_oid(NameStr(agg->data.user_view_schema), false));
+	Relation cagg_view_rel = table_open(cagg_view_oid, AccessShareLock);
+	RuleLock *cagg_view_rules = cagg_view_rel->rd_rules;
+	Assert(cagg_view_rules && cagg_view_rules->numLocks == 1);
+	RewriteRule *rule = cagg_view_rules->rules[0];
+	if (rule->event != CMD_SELECT)
+		elog(ERROR, "rule missing for view ");
```

Comment:
The code path needs a corresponding test case, if it's possible to add one. Otherwise, please add a comment that it should never happen in practice and is needed only for debugging purposes.

Reply:
modified error to use unexpected error code to make it obvious that it is not possible.